walk-in-interview/src/main/webapp/applicant-interested-list/script.js

```diff
@@ -0,0 +1,193 @@
+}
+
+/**
+ * Makes GET request to retrieve all the interested job listings
+ * belonging to the current applicant user. 
+ * This function is called when the interest page is loaded.
+ * 
+ * @param {int} pageSize The number of jobs for one page.
+ * @param {int} pageIndex The page index (starting from 0).
+ * @return {Object} The data returned from the servlet.
+ */
+function getInterestedJobs(pageSize, pageIndex) {
+  const params = `pageSize=${pageSize}&pageIndex=${pageIndex}`;
+    
+  fetch(`${API['applicant-interested-list']}?${params}`, {
+    method: 'GET',
+    headers: {'Content-Type': 'application/json'},
+  })
+      .then((response) => response.json())
+      .then((data) => {
+        console.log('data', data);
+        // Resets the error (there might have been an error msg from earlier)
+        setErrorMessage(/* errorMessageElementId= */ 'error-message',
+            /* msg= */ '', /* includesDefaultMsg= */ false);
+        return data;
+      });
+}
+
+/**
+ * This will add all the interested job listings onto the page.
+ * 
+ * @param {Object} jobPageData The details to be shown on the page.
+ */
+function displayInterestedJobListings(jobPageData) {
+  const jobListingsElement = document.getElementById('job-listings');
+  const jobShowing = document.getElementById('job-listings-showing');
+
+  /* reset the list so we don't render the same jobs twice */
+  jobListingsElement.innerHTML = '';
+  jobShowing.innerHTML = '';
+
+  if (jobPageData === undefined ||
+    !jobPageData.hasOwnProperty('jobList') ||
+    jobPageData['jobList'].length === 0) {
+    setErrorMessage(/* errorMessageElementId= */ 'error-message',
+        /* msg= */ STRINGS['no-jobs-error-message'],
+        /* includesDefaultMsg= */ false);
+    return;
+  }
+
+  const jobListings = jobPageData['jobList'];
+
+  jobListings.forEach((job) => {
+    jobListingsElement.appendChild(buildJobElement(job));
+  });
+
+  jobShowing.innerText = `${jobPageData['range'].minimum} -` +
+    ` ${jobPageData['range'].maximum} ${STRINGS['job-listings-showing']} ` +
+    `${jobPageData['totalCount']}`;
+}
+
+/**
+ * Builds the job element given the job details from the servlet response.
+ * 
+ * @param {Object} job The job to be displayed.
+ * @return {Element} The job listing element.
+ */
+function buildJobElement(job) {
+  const jobPostPreviewTemplate =
+    document.getElementById('job-listing-template');
+
+  const jobPostPreview =
+    jobPostPreviewTemplate.cloneNode( /* and child elements */ true);
+  jobPostPreview.setAttribute('id', job['jobId']);
+
+  const jobTitle = jobPostPreview.children[0];
+  jobTitle.innerText = job['jobTitle'];
+
+  const jobAddress = jobPostPreview.children[1];
+  const location = job['jobLocation'];
+  jobAddress.innerText = `${location['address']}, ${location['postalCode']}`;
+
+  const jobPay = jobPostPreview.children[2];
+  const pay = job['jobPay'];
+  jobPay.innerText = `${pay['min']} - ${pay['max']} SGD ` +
```

Comment:
Typically you don't construct things like that in code because of localization (translation). You define the strings with arguments in your string file and inject the argument values when rendering. Something like (in pseudo code):
```
Strings.js:
{
   [...]
   "jobPayDescription": "{MIN_PAY} - {MAX_PAY} {CURRENCY} (FREQUENCY)",
}
[...]
jobPay.innerText = String.format(Strings["jobPayDescription"], { 
      MIN_PAY: pay['min'],
      MAX_PAY: pay['min'],
      CURRENCY: currency,
     FREQUENCY: Strings["PaymentFrequency"][pay['paymentFrequency']]
});
```

Also note how i'm not rendering the enum for payment frequency as-is, i run it through the Strings to convert enum to string.

Reply:
You might want to address the other strings in this file like jobAddress, jobShowing or requirements list.